lib/utils/atRuleRawsAfterIndex.mjs

```diff
@@ -0,0 +1,7 @@
+/**
+ * @param {import('postcss').AtRule} atRule
+ * @returns {number}
+ */
+export default function atRuleRawsAfterIndex(atRule) {
+	return atRule.source?.end?.offset ?? atRule.toString().length;
```

Comment:
refactor
minor

This looks similar to [`return rule.source?.end?.offset ?? rule.toString().length;`](https://github.com/stylelint/stylelint/pull/8066/files#diff-15dd133c5fee2346a62b4e8f9ab0b5da1430f6008a4080cd9c885e9aa4c5fee8R6).
Could you have only one function which accept both node types?
i.e.
```
@import {AtRule, Rule} from 'postcss'
…
@param {AtRule | Rule} node
```

Reply:
> the actual implementation in PostCSS is very different for each node type

Interesting.

> but this would need a very specific name

If you pass a `Node` instead of `Rule | AtRule` it could be named `nodeRawsAfterIndex`.

> I am unsure.

Ill leave the decision to you.
We normally don't add that many one-line utils in a single PR so I was trying to lower that number.

Reply:
> If you pass a `Node` instead of `Rule | AtRul`e it could be named `nodeRawsAfterIndex`.

I think this is the best option, but implies even more tests 😛 
Then we also need coverage for `Declaration`, `Root` and `Document`

I agree that too many small helpers aren't ideal.

Reply:
To find the name, just try to add a JSDoc @description; that will help in the process.
Else just don't refactor and resolve this thread.

Reply:
When looking at this again I noticed that I misunderstood the `raws.after` field.

I thought this was about this:

```css
.foo {}  .bar{}
/*     ↑↑       */
```

But it is not, it is about this:


```css
.foo { color: blue;  }
/*                 ↑↑   */
```

So my previous implementation was incorrect, which also meant that we didn't have any test coverage for `raws.after` in the rule itself.

I've fixed the implementation and added more test coverage.

-----

I ultimately decided to keep these as separate functions because I am unsure if I understand it well enough at this time to pick the right abstraction.

Reply:
> I ultimately decided to keep these as separate functions because I am unsure if I understand it well enough at this time to pick the right abstraction.

OK.

> I've fixed the implementation and added more test coverage.

Can you add another changeset.
This fix is not about reported ranges per se.
i.e. false negative

Reply:
> Can you add another changeset.
> This fix is not about reported ranges per se.

It is still purely about reported ranges :)
Before this change we used `word` in report, which also gave ok results.
It just wasn't tested.

Reply:
[Demo for `raws.after` with the current Stylelint version](https://stylelint.io/demo/#N4Igxg9gJgpiBcIB0AzCEAEwOQDYQCd4MAjXAVxgG4AdFAXxABoQUBLXGAOQEMBbOIhgAPfgAdOSMAGdpzcBAB27AOYIQwGoowYaIEQBcYiqNL3EA2lp0690gwE9OuNooMBaSMrYr39niY8BFB61hgAulqMLF6qAGKEfDwG6gBW0krysGJyiJrauiD2TjAubuaFuMkw9npMYXaOzq4esT5+BgFQQSEIldW1IFEg9EA)